src/main.rs

```diff
@@ -0,0 +1,72 @@
+use std::env;
+use std::process;
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::Instant;
```

Comment:
since we're not printing wall-clock time, time::precise_time_ns() is faster (citation needed)

Reply:
Looks like `std::time::Instant` is dependable in this regard now: https://stackoverflow.com/questions/55583503/is-stdtimeduration-as-precise-as-timeprecise-time-ns-from-time-crate

Reply:
Would be good to add some examples that measure overhead of various time calls. For now no strong objection to using Instant::now() esp since it's not being called frequently.